script/draft-release/version.ts

```diff
@@ -0,0 +1,57 @@
+import { inc, parse } from 'semver'
+
+import { Channel } from './channel'
+
+export function getNextVersionNumber(
+  version: string,
+  channel: Channel
+): string {
+  const parsed = parse(version)
+
+  if (parsed == null) {
+    throw new Error(`Unable to parse input '${version}' into version`)
+  }
+
+  if (channel === 'production') {
+    if (parsed.prerelease.some(p => p.startsWith('beta'))) {
+      throw new Error(
+        `Unable to resolve production version using beta release '${version}'`
+      )
+    }
+
+    if (parsed.prerelease.some(p => p.startsWith('test'))) {
+      throw new Error(
+        `Unable to resolve production version using test release '${version}'`
+      )
+    }
+
+    const nextVersion = inc(version, 'patch')
+
+    if (nextVersion == null) {
+      throw new Error(
+        `Unable to resolve next version from release '${version}'`
+      )
+    }
+
+    return nextVersion
+  }
+
+  if (parsed.prerelease.some(p => p.startsWith('test'))) {
+    throw new Error(
+      `Unable to resolve production version using test release '${version}'`
+    )
+  }
+
+  const betaTagIndex = version.indexOf('-beta')
```

Comment:
This can be extracted into a function, `tryIncrementBetaVersion: (version: string) => string` (or similar name)?